src/cmdstan/command.hpp

```diff
@@ -264,16 +267,12 @@ int command(int argc, const char *argv[]) {
           save_iterations, refresh, interrupt, logger, init_writer,
           sample_writers[0], diagnostic_json_writers[0]);
     } else {
-      std::string output_file
-          = get_arg_val<string_argument>(parser, "output", "file");
-      auto base_sfx = get_basename_suffix(output_file);
-      if (base_sfx.second.empty())
-        base_sfx.second = ".csv";
-      auto ofs = std::make_unique<std::ofstream>(base_sfx.first + "_pathfinder"
-                                                 + base_sfx.second);
-      if (sig_figs > -1) {
+      std::vector<std::string> pf_names = make_filenames(
+          get_arg_val<string_argument>(parser, "output", "file"), "", ".csv", 1,
+          id);
+      auto ofs = std::make_unique<std::ofstream>(pf_names[0]);
```

Comment:
I think I still prefer the original code which just repeated the piece of logic needed rather than calling this function and creating a length-1 vector